util/store/iter_test.go

```diff
@@ -0,0 +1,37 @@
+package store
+
+import (
+	"testing"
+
+	"github.com/umee-network/umee/v4/tests/tsdk"
+	"gotest.tools/v3/assert"
+)
+
+func TestIterate(t *testing.T) {
+	type pair struct {
+		K, V []byte
+	}
+	pairs := []pair{
+		{[]byte{0}, []byte{0}},
+		{[]byte{1, 1}, []byte{0}},
+		{[]byte{1, 7}, []byte{3}},
+		{[]byte{1, 8}, []byte{2}},
+		{[]byte{2, 4}, []byte{1}},
+	}
+
+	db := tsdk.NewKVStore(t)
+	db.Set(pairs[0].K, pairs[0].V)
+	db.Set(pairs[4].K, pairs[4].V)
+	db.Set(pairs[3].K, pairs[3].V)
+	db.Set(pairs[1].K, pairs[1].V)
+	db.Set(pairs[2].K, pairs[2].V)
```

Comment:
for / range loop might make sense here, or a comment indicating why it won't (e.g. `pairs` is sorted but we want to `db.Set` the keys out of order to properly test iterator)

Reply:
BTW, would be great if you add more edge cases and also add a test to the reverse iterator. I'm also planning to write a generic fetcher, but that will be a new feature.

Reply:
@toteki I will create a task to add missing tests. This PR is meant to provide the helper functions I mentioned with example how to use it.